numba/typing/templates.py

```diff
@@ -426,9 +431,38 @@ def generic(self, args, kws):
             return
         # Compile and type it for the given types
         disp_type = types.Dispatcher(disp)
-        sig = disp_type.get_call_type(self.context, args, kws)
-        # Store the compiled overload for use in the lowering phase
-        self._compiled_overloads[sig.args] = disp_type.get_overload(sig)
+        # Store the compiled overload for use in the lowering phase if there's
+        # no inlining required (else functions are being compiled which will
+        # never be used as they are inlined)
+        if self._inline != 'never':
```

Comment:
Maybe make `self._inline` an `InlineOptions`.  And also adding helper methods like `.is_never_inline()`, `.is_always_inline()`, `.has_cost_model()` so that it doesn't need to know how each option is spelled exactly.

Reply:
good idea, will change.